share/p2p/peers/manager.go

```diff
@@ -413,7 +417,9 @@ func (m *Manager) blacklistPeers(reason blacklistPeerReason, peerIDs ...peer.ID)
 		return
 	}
 	for _, peerID := range peerIDs {
-		m.fullNodes.remove(peerID)
+		if !m.disc.Discard(peerID) {
+			m.fullNodes.remove(peerID)
```

Comment:
wait so only remove from full node pool if NOT discarded successfully by disc?

Reply:
only remove from fn pool if the peer was not from discovery. This is because, if the peer was from discovery, it will remove it from fn pool through the WithOnPeersUpdate